deepchem/feat/graph_features.py

```diff
@@ -398,12 +400,79 @@ def bond_features(bond, use_chirality=False):
   ]
   if use_chirality:
     bond_feats = bond_feats + one_of_k_encoding_unk(
-        str(bond.GetStereo()), possible_bond_stereo)
+        str(bond.GetStereo()), GraphConvCoonstants.possible_bond_stereo)
   return np.array(bond_feats)
 
 
-def pair_features(mol, edge_list, canon_adj_list, bt_len=6,
-                  graph_distance=True):
+def max_pair_distance_pairs(mol: RDKitMol,
+                            max_pair_distance: Union[int, str]) -> np.ndarray:
+  """Helper method which finds atom pairs within max_pair_distance graph distance.
+
+  This helper method is used to find atoms which are within max_pair_distance
+  graph_distance of one another. This is done by using the fact that the
+  powers of an adjacency matrix encode path connectivity information. In
+  particular, if `adj` is the adjacency matrix, then `adj**k` has a nonzero
+  value at `(i, j)` if and only if there exists a path of graph distance `k`
+  between `i` and `j`. To find all atoms within `max_pair_distance` of each
+  other, we can compute the adjacency matrix powers `[adj, adj**2,
+  ...,adj**max_pair_distance]` and find pairs which are nonzero in any of
+  these matrices. Since adjacency matrices and their powers are positive
+  numbers, this is simply the nonzero elements of `adj + adj**2 + ... +
+  adj**max_pair_distance`.
+
+  Parameters
+  ----------
+  mol: rdkit.Chem.rdchem.Mol
+    RDKit molecules
+  max_pair_distance: Union[int, str], (default 'infinity')
+    This value can be a positive integer or the string 'infinity'. This
+    parameter determines the maximum graph distance at which pair features
+    are computed. For example, if `max_pair_distance==2`, then pair features
+    are computed only for atoms at most graph distance 2 apart.
+
+
+  Returns
+  -------
+  np.ndarray
+    Of shape `(2, num_pairs)` where `num_pairs` is the total number of pairs
+    within `max_pair_distance` of one another.
+  """
+  from rdkit import Chem
+  from rdkit.Chem import rdmolops
+  N = len(mol.GetAtoms())
+  if (isinstance(max_pair_distance, str) and
+      max_pair_distance == "infinity") or (isinstance(max_pair_distance, int)
+                                           and max_pair_distance >= N):
+    max_distance = N
+  elif (
+      (isinstance(max_pair_distance, int) and max_pair_distance <= 0) or
+      (isinstance(max_pair_distance, str) and max_pair_distance != "infinity")):
+    raise ValueError(
+        "max_pair_distance must either be a positive integer or the string 'infinity'"
+    )
+  elif isinstance(max_pair_distance, int):
+    max_distance = max_pair_distance
+  adj = rdmolops.GetAdjacencyMatrix(mol)
+  # Handle edge case of self-pairs (i, i)
+  sum_adj = np.eye(N)
+  for i in range(max_distance):
+    # Increment by 1 since we don't want 0-indexing
+    power = i + 1
+    sum_adj += np.linalg.matrix_power(adj, power)
```

Comment:
This is a slow way of computing it, though possibly it's still fast enough that it doesn't matter.